pkg/apis/autoscaling/validation/validation.go

```diff
@@ -344,8 +344,9 @@ func validateObjectSource(src *autoscaling.ObjectMetricSource, fldPath *field.Pa
 	allErrs = append(allErrs, validateMetricIdentifier(src.Metric, fldPath.Child("metric"))...)
```

Comment:
/hold

standard questions when we want to tighten validation:
* does it prevent a no-op update of existing persisted objects that would fail this new validation? (would block storage migration tasks)
* does it prevent deletion of existing persisted objects with finalizers that would fail this new validation? (update is required to remove finalizers to allow deletion)
* does it prevent controllers from updating status of existing persisted objects that would fail this new validation?

we could consider ratcheting validation:
* apply this tightened validation on create
* apply this tightened validation on update where the old object passed this validation (don't let a valid object become invalid)
* skip this tightened validation on update where the old object fails this validation

c.f. https://github.com/kubernetes/kubernetes/issues/59984

Reply:
If we do tighten validation using ratcheting, it would look like:

1. in `pkg/apis/.../validation/validation.go`, add an `Allow... bool` field to `...ValidationOptions`
2. make sure that options struct is passed through all validation functions down to this location
3. in this function, only allow the relaxed validation if the `Allow...` field is true
4. in `GetValidationOptionsFrom...`, compute the validation options based on the new and old object
    * in this release
      * on create (new object is non-nil, old object is nil), set the `Allow...` field to `false`
      * on update, set the `Allow...` field to `true` *only* if the old object already has ...
    * in the next release, set the `Allow...` field to `true` unconditionally

Reply:
Can we also use the warning mechanism at this point and still give feedback to the user that something isn't right with their object?